domain/domain.go

```diff
@@ -1302,26 +1401,34 @@ func (do *Domain) runawayRecordFlushLoop() {
 			} else if fired {
 				fired = false
 				// meet a new record, reset the timer.
-				timer.Reset(time.Second)
+				runawayRecordFluashTimer.Reset(time.Second)
+			}
+		case <-runawayRecordGCTicker.C:
+			if owner.IsOwner() {
+				do.gcSystemTable("tidb_runaway_queries", "time", runawayRecordExpiredDuration)
```

Comment:
As `gcSystemTable` can be very slow, to avoid blocking handling other events, better spawn a new routine to do this job.